documentation/docs/05-misc/04-migrating-v3-to-v4.md

```diff
@@ -2,4 +2,56 @@
 title: Migrating from v3 to v4
 ---
 
-TODO
+This migration guide provides an overview of how to migrate from Svelte version 3 to 4. See the linked PRs for more detailed migration instructions. Use the migration script to migrate some of these automatically: `npx svelte-migrate svelte-4`
+
+If you're a library author, consider whether to only support Svelte 4 or if it's possible to support Svelte 3, too, after migrating - since most breaking changes don't affect many people, this may be easily possible. Also remember to update your `peerDependencies`.
+
+## Minimum version requirements
+
+- Upgrade to Node 16 or higher. Earlier versions are no longer supported. ([#8566](https://github.com/sveltejs/svelte/issues/8566))
+- If you are using Vite, the minimum supported vite-plugin-svelte version is now 2.4.1. If you are a SvelteKit user, upgrade to 1.15.9 or newer to ensure compatibility. ([#8516](https://github.com/sveltejs/svelte/issues/8516))
+- If you are using webpack, upgrade to webpack 5 or higher. Earlier versions are no longer supported. ([#8515](https://github.com/sveltejs/svelte/issues/8515))
+- If you are using TypeScript, upgrade to TypeScript 5 or higher. Lower versions might still work, but no guarantees are made about that. ([#8488](https://github.com/sveltejs/svelte/issues/8488))
+
+## Browser conditions for bundlers
+
+Bundlers must now specify the browser condition when building a frontend bundle for the browser. SvelteKit and Vite will handle this automatically for you, for Rollup or webpack you may need to adjust your config. ([#8516](https://github.com/sveltejs/svelte/issues/8516))
+
+## Removal of CJS related output
+
+Svelte no longer supports the CommonJS (CJS) format for compiler output and has also removed the `svelte/register` hook and the CJS runtime version. If you needed those, consider using a bundler like Vite or the full-stack framework SvelteKit instead. ([#8613](https://github.com/sveltejs/svelte/issues/8613))
```

Comment:
it would be a lot of work to migrate a non-sveltekit project to sveltekit, so it feels like a harsh migration suggestion. how about this?
```suggestion
Svelte no longer supports the CommonJS (CJS) format for compiler output and has also removed the `svelte/register` hook and the CJS runtime version. If you needed those, consider setting `"type": "module"` in your project and using ESM. This has been set upon creating a new Svelte project for the past couple of years and most of the Svelte ecosystem now expects it. ([#8613](https://github.com/sveltejs/svelte/issues/8613))
```

Reply:
webpack and rollup set `esm` by default. What I imagined was someone not using a bundler at all, instead handrolling something, which is why they need to cjs output for some reason. You're right that someone could use cjs together with a bundler already. I'm not sure why they would want to do that and what would be needed to migrate, which is why I'd like to keep it rather vague.